Automattic/wp-calypso
# Components: remove redundant ProductPurchaseFeatures component from blocks.

Part of fixing https://github.com/Automattic/wp-calypso/issues/22691.

`ProductPurchaseFeatures` is used solely as a wrapper to sub-components rendering plans feature, introducing no extra functionality, nor styling.

Other changes:
- move the ProductPurchaseFeaturesList higher up the folder structure as it was a sole sub-folder of `/product-purchase-features`
- replace `ProductPurchaseFeatures` call with a div in current plan feature page.

**To test:**
- checkout this branch in your local Calypso
- verify that `http://calypso.localhost:3000/plans/my-plan/:site` is unchanged for any .com or Jetpack connected site.
